File: Android/TestSuperModel/app/src/main/java/com/soonzik/testsupermodel/News.java
package com.soonzik.testsupermodel;

import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Date;

/**
 * Created by kevin_000 on 21/03/2015.
 */
public class News extends ActiveRecord {
    private int id = -1;
    private String title = "";
    private Date date = null;
    private String news_type = "";
    private User user = null;
    private ArrayList<Newstext> newstexts = null;
    private ArrayList<Attachment> attachments = null;
    private ArrayList<Tag> tags = null;
    public News() {}

    public News(JSONObject json) {
        super.createInstance(this, json, this.getClass());
    }

    @Override
    public String toString() {
        String str = "id = " + Integer.toString(id)
                    + " : author = { " + (user != null ? user.toString() : "") + " }"
                    + " : date = " + (date != null ? date.toString() : "")
                    + " : newstexts = [";
        if (newstexts != null) {
            for (Newstext newstext : newstexts) {
                str += "{ " + newstext.toString() + " } ";
            }
        }
        str += " ]";
        str += " : attachments = [ ";
        if (attachments != null) {
            for (Attachment attachment : attachments) {
                str += "{ " + attachment.toString() + " } ";
            }
        }
        str += " ]";
        return (str);
    }

    public ArrayList getNewstext() {
        return newstexts;
    }

    public String getTitle() {
        return title;
    }

    public int getId() {
        return id;
    }
}

File: iOS SoonZik/SoonZik/Models/News.m
//
//  News.m
//  SoonZik
//
//  Created by LLC on 08/07/2014.
//  Copyright (c) 2014 Coordina. All rights reserved.
//

#import "News.h"
#import "Attachment.h"
#import "NewsText.h"

@implementation News

- (id)initWithJsonObject:(NSDictionary *)json
{
    self = [super init];
    NSLog(@"NEWS JSON : %@", json);
    
    self.identifier = [[json objectForKey:@"id"] intValue];
    self.title = [json objectForKey:@"title"];
    self.type = [json objectForKey:@"news_type"];
    
    NSDateFormatter *dateformat = [[NSDateFormatter alloc] init];
    [dateformat setDateFormat:@"yyyy-MM-dd HH:mm:ss"];
    dateformat.timeZone = [NSTimeZone timeZoneForSecondsFromGMT:1];
    self.date = [dateformat dateFromString:[json objectForKey:@"created_at"]];
    
    self.listOfAttachments = [[NSMutableArray alloc] init];
    if ([[json objectForKey:@"attachments"] count] > 0) {
        for (NSDictionary *dict in [json objectForKey:@"attachments"]) {
            Attachment *atta = [[Attachment alloc] initWithJsonObject:dict];
            [self.listOfAttachments addObject:atta];
        }
    }
    
    NSString *language = [[NSLocale preferredLanguages] objectAtIndex:0];
    NSString *lang;
    if ([language isEqualToString:@"en"]) {
        lang = @"EN";
    } else if ([language isEqualToString:@"fr"]) {
        lang = @"FR";
    }
    
    self.listOfNewsTexts = [[NSMutableArray alloc] init];
    for (NSDictionary *dict in [json objectForKey:@"newstexts"]) {
        if ([lang isEqualToString:[dict objectForKey:@"language"]]) {
            NewsText *newsText = [[NewsText alloc] initWithJsonObject:dict];
            [self.listOfNewsTexts addObject:newsText];
        }
    }
    
    return self;
}

- (void)encodeWithCoder:(NSCoder *)aCoder{
    [aCoder encodeObject:[NSString stringWithFormat:@"%i", self.identifier] forKey:@"identifier"];
    [aCoder encodeObject:self.title forKey:@"title"];
    [aCoder encodeObject:self.listOfContents forKey:@"listOfContents"];
    [aCoder encodeObject:self.author forKey:@"author"];
    [aCoder encodeObject:self.type forKey:@"type"];
    [aCoder encodeObject:self.date forKey:@"date"];
    [aCoder encodeObject:self.listOfAttachments forKey:@"listOfAttachments"];
    [aCoder encodeObject:self.listOfNewsTexts forKey:@"listOfNewsTexts"];
    [aCoder encodeObject:self.listOfComments forKey:@"listOfComments"];
}

- (id)initWithCoder:(NSCoder *)aDecoder{
    if(self = [super init]){
        self.identifier = [[aDecoder decodeObjectForKey:@"identifier"] intValue];
        self.title = [aDecoder decodeObjectForKey:@"title"];
        self.listOfComments = [aDecoder decodeObjectForKey:@"listOfComments"];
        self.author = [aDecoder decodeObjectForKey:@"author"];
        self.type = [aDecoder decodeObjectForKey:@"type"];
        self.date = [aDecoder decodeObjectForKey:@"date"];
        self.listOfAttachments = [aDecoder decodeObjectForKey:@"listOfAttachments"];
        self.listOfNewsTexts = [aDecoder decodeObjectForKey:@"listOfNewsTexts"];
        self.listOfComments = [aDecoder decodeObjectForKey:@"listOfComments"];
    }
    return self;
}

@end
